halva/src/main/java/io/soabase/halva/comprehension/FutureFor.java

```diff
@@ -41,10 +41,6 @@
         _delegate.letComp(any, let_supplier);
         return this;
     }
-    public FutureFor filter(final Supplier<Boolean> test) {

```

Comment:
The implications of removing `filter`? None, that's just not a supported operation in for comprehensions for many monads (in scala you'll get a syntax error if you insert an `if` statement, this fix was just making halva in line with that instead of throwing a runtime exception).

It won't have any implications on the let comprehension (or did i misunderstand that question?)